esmf_regrid/schemes.py

```diff
@@ -575,29 +1051,186 @@ def __call__(self, cube):
             area-weighted regridding via :mod:`esmpy` generated weights.
 
         """
-        src_x, src_y = (_get_coord(cube, "x"), _get_coord(cube, "y"))
-
-        # Check the source grid matches that used in initialisation
-        if self.src_grid != (src_x, src_y):
-            raise ValueError(
-                "The given cube is not defined on the same "
-                "source grid as this regridder."
-            )
+        if cube.mesh is not None:
+            # src_mesh = cube.mesh
+            # location = cube.location
+            # if self.src != (src_mesh, location):
+            #     raise ValueError(
+            #         "The given cube is not defined on the same "
+            #         "source mesh as this regridder."
+            #     )
+            dims = [cube.mesh_dim()]
 
-        if len(src_x.shape) == 1:
-            grid_x_dim = cube.coord_dims(src_x)[0]
-            grid_y_dim = cube.coord_dims(src_y)[0]
         else:
-            grid_y_dim, grid_x_dim = cube.coord_dims(src_x)
+            src_x, src_y = (_get_coord(cube, "x"), _get_coord(cube, "y"))
 
-        regrid_info = RegridInfo(
-            x_dim=grid_x_dim,
-            y_dim=grid_y_dim,
-            x_coord=self.grid_x,
-            y_coord=self.grid_y,
+            # Check the source grid matches that used in initialisation
+            if self.src != (src_x, src_y):
+                raise ValueError(
+                    "The given cube is not defined on the same "
+                    "source grid as this regridder."
+                )
+
+            if len(src_x.shape) == 1:
+                dims = [cube.coord_dims(src_x)[0], cube.coord_dims(src_y)[0]]
+            else:
+                dims = cube.coord_dims(src_x)[::-1]
```

Comment:
Yes, see https://github.com/SciTools-incubator/iris-esmf-regrid/pull/198/commits/e303d767f621c97b5d7d3de384eb5b55950b2eae . I had previously attempted an alternative solution but I've decided to keep the more recent solution to this since it seems more comprehensive. This effectively keeps the lazy data handling the same as it was before this PR, though I've expanded the tests just to be sure.

Reply:
>Yes, see e303d767f621c97b5d7d3de384eb5b55950b2eae

I think we might be talking about two different code blocks that happen to look similar?

Reply:
I guess it doesn't help that I asked two different questions in the same thread!

Reply:
I believe dims is still iterable in both cases, note that this is slicing an iterable rather than indexing it.

Reply:
OK, my bad! I consider this thread resolved.